sklearn/preprocessing/_discretization.py

```diff
@@ -244,20 +278,38 @@ def transform(self, X):
         Xt : numeric array-like or sparse matrix
             Data in the binned space.
         """
+        X = check_array(X, dtype='numeric')
+        n_samples, n_features = X.shape
+
+        if self.subsample == 'warn':
+            warnings.warn("In the future (v0.22) onwards subsample = 1e5"
+                          "will be used by default. Pass subsample=None to"
+                          "silence this warning for now.", FutureWarning)
+
+        if (self.subsample is not None) and (self.subsample != 'warn'):
+
+            if n_samples > self.subsample:
+                subsample_idx = self.random_state.choice(
+                                n_samples, size=self.subsample,
+                                replace=False)
+                X = X.take(subsample_idx, mode='clip')
+
         check_is_fitted(self, ["bin_edges_"])
 
         Xt = check_array(X, copy=True, dtype=FLOAT_DTYPES)
         n_features = self.n_bins_.shape[0]
         if Xt.shape[1] != n_features:
             raise ValueError("Incorrect number of features. Expecting {}, "
-                             "received {}.".format(n_features, Xt.shape[1]))
+                             "received {}.".format(n_features,
```

Comment:
revert this change

Reply:
I have done it. How do I resolve this issue ?

>>> est = KBinsDiscretizer(n_bins=3, encode='ordinal', strategy='uniform',
UNEXPECTED EXCEPTION: SyntaxError('unexpected EOF while parsing', ('<doctest sklearn.preprocessing._discretization.KBinsDiscretizer[1]>', 1, 71, "est = KBinsDiscretizer(n_bins=3, encode='ordinal', strategy='uniform',\n"))
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/testenv/lib/python3.7/doctest.py", line 1329, in __run
    compileflags, 1), test.globs)
  File "<doctest sklearn.preprocessing._discretization.KBinsDiscretizer[1]>", line 1
    est = KBinsDiscretizer(n_bins=3, encode='ordinal', strategy='uniform',
                                                                         ^
SyntaxError: unexpected EOF while parsing

Reply:
EOF while parsing generally means you did not close a parenthesis.
In the doctest, you need to insert `...` when going to next line, otherwise the doctest interprets a new line as the end of the instruction.

See how I modified:
```
    >>> est = KBinsDiscretizer(n_bins=3, encode='ordinal', strategy='uniform',
    subsample=None)
```
into
```
    >>> est = KBinsDiscretizer(n_bins=3, encode='ordinal', strategy='uniform',
    ...                        subsample=None)
```